### DISCUSSION TITLE
AudioClip notifications

### DISCUSSION THREAD

**Discussion by svrooij** [svrooij/sonos2mqtt#191] [General]
Sending an audio notification has been possible for years with this app. Since the latest beta you use the notifications using the native **Audio Clip** endpoint. This used to be cloud only, but someone discovered it to be no longer cloud only.

We are not ready to switch this for everybody, but you can try it out by starting Sonos2Mqtt with the `SONOS2MQTT_EXPERIMENTAL` environment variable set to `true`. This will switch the `notify` and the `speak` command to use the Audio Clip endpoint. You can also use the `--experimental` flag if you're still running this app from the command line.

Please provide your feedback below.

**Comment by jockele79** [comment]
Hi,

first of all sorry for my late response. I just wanted to test the experimental version, but I'm not sure if it was correctly activated.
I sent :
`sonos/RINCON_F0F6C189D33001400/control
{
  "command": "SONOS2MQTT_EXPERIMENTAL",
  "input": "true"
}`

Result within the logfile:
` "command": "SONOS2MQTT_EXPERIMENTAL",
  "input": "true"
} to topic sonos/RINCON_F0F6C189D33001400/control (QoS 0, Retained: false)
2024-04-29 18:49:26,034  INFO --- MqttFX ClientModel             : messageArrived() with topic: sonos/RINCON_F0F6C189D33001400/control
2024-04-29 18:49:26,036  INFO --- MqttFX ClientModel             : messageArrived() added: message #5 to topic 'sonos/RINCON_F0F6C189D33001400/control' `

According to your mail a message within the logfile should appear:
2024-02-07T08:05:35.298Z [Information] Experimental features activated, please provide feedback on notifications. https://github.com/svrooij/sonos2mqtt/discussions/191

Was the activation sent correctly?

**Reply by svrooij** [reply]
Are you running sonos2mqtt in docker? Then you’ll need to add that variable to the environment variables. Are you running it from the command line (I hope not) the. You should add the start argument —experimental

**Comment by jockele79** [comment]
Hi,

I'm running it in docker, managed by Portainer. I added the variable, see screenshot:

![image](https://github.com/svrooij/sonos2mqtt/assets/59440372/dddae807-7be3-4aaa-ab7f-3d850ccd23d7)

Unfortunately I don't think that the function is activated, log looks like this:

`      
2024-05-05T07:18:44.671Z [Information] Starting sonos2mqtt v3.2.0


2024-05-05T07:18:44.798Z [Information] LogLevel changed to information


2024-05-05T07:18:44.918Z [Information] Found 2 sonos speakers


2024-05-05T07:18:45.526Z [Information] Mqtt connection changed to connected: true

`

Any idea what else I could try?

**Reply by svrooij** [reply]
What is the version of the app you're running?

Should be any of the beta versions.
https://github.com/svrooij/sonos2mqtt/releases/tag/v3.2.0-beta.19

You should set the tag to it, so not using latest but this one.

`FROM ghcr.io/svrooij/sonos2mqtt:beta` or `FROM ghcr.io/svrooij/sonos2mqtt:3.2.0-beta.19`

**Comment by mitch-geht-ab** [comment]
Hey @svrooij 
have enabled experimental for the PlayNotificationAudioClip feature in beta for the notify command.
https://github.com/svrooij/sonos2mqtt/commit/0b0b4010471b992a29c011dc6b354358bc4a1e0b
and it is awesome!
works much better than the old-notify implementation.
had the problem that the old-notify hasn't work with start stream from the spotify app.
music stopped, notification track was played but restoring the spotify stream wasn't possible.
worked when you start your track via sonos app (and spotify music).

I guess difference was in
```
# started from spotify app
"CurrentTrackURI": "x-sonos-spotify:spotify%3atrack%3a14pJxK2NThPFEQYNGfMSGF?sid=9&amp;flags=8232&amp;sn=14",
# started from sonos app 
"CurrentTrackURI": "x-sonos-spotify:spotify:track:5PFx3hjH2REjYpbDqzEXaA?sid=9&amp;flags=0&amp;sn=14",
```
maybe the flag is the magic or it is something else. if you are interested I can send you a full "sonos/status/sonosboxxy/avtransport" json.

anyway... with overlay notify everything is working like a charm.

**Reply by svrooij** [reply]
I might change this to be the default behavior and then only use the old implementation when it detected an S1 system
